### ISSUE TITLE
Parallel migration support

### ISSUE THREAD

**Issue by xiaoyao9184** [xataio/pgroll#208]
Since the migration process is divided into `start` and `complete`,
the migration process may need weeks or months, or not run `complete` forever.

Then it is very likely that the previous migration was not completed and the next migration was started.
I have tried do that with examples, but failed.

```
Failed to start migration: a migration for schema "public" is already in progress
```

I know this will be complicated, like git branching, but I think it will be very useful.

**Comment by HoodieCollin** [comment]
I agree that this would be very useful. My team works on multiple features in parallel during a development cycle. With multiple in-progress migrations support, we could have all these features coexist in our staging Env without having to worry about coordinating while developing.

**Comment by alfari16** [comment]
looking forward for this feature 👀

**Comment by akashwar** [comment]
We are exploring this tool for our use case of zero downtime migration. It is important for us to get this issue resolved before we start using your tool.

Can you guys let us know the timeline you are targeting for fixing this issue in your v1 milestone?
@andrew-farries

**Comment by gulcin** [comment]
Hello @akashwar glad to hear you're considering pgroll. Unfortunately, we don't have a timeline for this issue yet, but we'll keep you posted when we prioritize it again.
